subprojects/core/src/main/java/org/gradle/api/tasks/JavaExec.java

```diff
@@ -528,4 +532,14 @@ public boolean isIgnoreExitValue() {
     public List<CommandLineArgumentProvider> getJvmArgumentProviders() {
         return javaExecHandleBuilder.getJvmArgumentProviders();
     }
+
+    /**
+     * Configure system properties.
+     *
+     * @since 4.7
+     */
+    @Incubating
+    public void systemProperties(Action<SystemPropertyHandler> action) {
+        action.execute(new DefaultSystemPropertyHandler(this));
```

Comment:
Newing one up every time doesn't seem like the right approach here. Do we need decoration for this type?

Reply:
Decoration doesn't seem necessary, WDYT?

Reply:
I changed the handler to be a field.